Uchu.World/Objects/Components/ReplicaComponents/MovingPlatformComponent.cs

```diff
@@ -148,56 +188,96 @@ public override void Serialize(BitWriter writer)
             }
         }
 
+        public void Stop()
+        {
+            Timer.Stop();
+            State = PlatformState.Idle;
+        }
+
+        public void MoveTo(uint position,Action moveCompleteCallback = default)
+        {
+            // Update Object in world.
+            CurrentWaypointIndex = position;
+            NextWaypointIndex = NextIndex;
+            _currentDuration = (WayPoint.Position - NextWayPoint.Position).Length() / WayPoint.Speed;
+            _wayPointStartTime = (DateTime.UtcNow - new DateTime(1970,1,1,0,0,0)).TotalSeconds;
+            State = PlatformState.Move;
+            TargetPosition = WayPoint.Position;
+            TargetRotation = WayPoint.Rotation;
+
+            GameObject.Serialize(GameObject);
+
+            // Start Waiting after completing path.
+            Timer = new Timer
+            {
+                AutoReset = false,
+                Interval = _currentDuration * 1000
+            };
+            Timer.Elapsed += (sender, args) =>
+            {
+                Timer.Stop();
+                
+                // Update Object in world.
+                CurrentWaypointIndex = NextIndex;
+                PathName = null;
+                State = PlatformState.Idle;
+                TargetPosition = WayPoint.Position;
+                TargetRotation = WayPoint.Rotation;
+                NextWaypointIndex = NextIndex;
+            
+                GameObject.Serialize(GameObject);
+            };
+            if (moveCompleteCallback != default)
+            {
+                Timer.Elapsed += (sender, args) => moveCompleteCallback();
+            }
+
+            Task.Run(() => Timer.Start());
```

Comment:
Pretty sure this doesn't have to run inside a Task.Run right? Timer.Start should exit immediately.

Reply:
That is correct. Pretty sure I started with a reason to do this but didn't need it.